### PR TITLE
Add support for arrays of different data types in the Trigger Form UI

### PR DESCRIPTION
**Pull request by MatthieuBlais**
### Description

The new Trigger Form UI released in 2.6 is handy but doesn't support `array` params of data types other than string. For more flexibility, this PR proposes to generate a JSON field similar to the `object` param if the users specify the attribute `items` in `schema`.

### How to reproduce the issue

Use an `array` param and require the items to be integers.

```
with DAG(
    dag_id=Path(__file__).stem,
    description=__doc__[0 : __doc__.find(".")],
    doc_md=__doc__,
    schedule=None,
    start_date=datetime.datetime(2022, 3, 4),
    catchup=False,
    tags=["example_ui"],
    params={
        "array_of_numbers": Param(
            [1, 2, 3],
            "Only integers are accepted in this array",
            type="array",
            title="Array of numbers",
            items={"type": "number"},
        ),
    },
) as dag:

    @task(task_id="show_params")
    def show_params(**kwargs) -> None:
        ti: TaskInstance = kwargs["ti"]
        dag_run: DagRun = ti.dag_run
        if not dag_run.conf:
            print("Uups, no parameters supplied as DagRun.conf, was the trigger w/o form?")
            raise AirflowSkipException("No DagRun.conf parameters supplied.")
        print(f"This DAG was triggered with the following parameters:\n{json.dumps(dag_run.conf, indent=4)}")

    show_params()
``` 

The Generated Conf JSON shows an array of strings

image

And we cannot trigger the DAG:

image

### Workaround

We can manually update the Generated Configuration JSON box, but it is not user friendly (especially if we expect an array of objects!), and will become counter intuitive as users get used to the benefits of the new trigger form.

What do you think @jens-scheffler-bosch ?

### CODE REVIEW

File: docs/apache-airflow/core-concepts/params.rst [hunk 1]
```diff
@@ -190,7 +190,8 @@ The following features are supported in the Trigger UI Form:
     - ``boolean``: Generates a toggle button to be used as ``True`` or ``False``.
     - ``date``, ``datetime`` and ``time``: Generate date and/or time picker
     - ``array``: Generates a HTML multi line text field, every line edited will be made into a string array as value.
-      if you add the attribute ``example`` with a list, a multi-value select option will be generated.
+      If you add the attribute ``example`` with a list, a multi-value select option will be generated.
+      If you add the attribute ``items``, a JSON entry field will be generated for more complex array types.
```
Comment (jscheffl) [inline docs/apache-airflow/core-concepts/params.rst]: Can you add a reference to JSON specs?
```suggestion
      If you add the attribute ``items``, a JSON entry field will be generated for more array types
      and additional type validation as described in
      `JSON Schema Array Items<https://json-schema.org/understanding-json-schema/reference/array.html#items>`_.
```

File: airflow/example_dags/example_params_ui_tutorial.py [hunk 2]
```diff
@@ -208,6 +216,18 @@
             title="JSON entry field",
             section="Special advanced stuff with form fields",
         ),
+        "array_of_objects": Param(
+            [{"name": "account_name", "country": "country_name"}],
+            "Array with complex objects and validation rules",
```
Comment (jscheffl) [inline airflow/example_dags/example_params_ui_tutorial.py]: A link to JSON Schema specs might be useful here as well.
```suggestion
            "Array with complex objects and validation rules. "
            "See <a href='https://json-schema.org/understanding-json-schema"
            "/reference/array.html#items'>JSON Schema validation options in specs.</a>",
```

**Review by jscheffl — APPROVED** [review]
Just some small beautification proposals in documentation, otherwise - I like it! Thanks for the contribution!
(Take my approval emotional, I have no committer status)

**Comment by MatthieuBlais** [comment]
Great, thanks for the feedback! I've updated the PR with your commits

**Comment by jens-scheffler-bosch** [comment]
> Great, thanks for the feedback! I've updated the PR with your commits

Uuups, sorry, my comments were not syntax checked :-( The URL is not correctly formed to be accepted by Sphinx.

**Review by jscheffl — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**